src/lib/logical_query_plan/lqp_utils.hpp

```diff
@@ -127,6 +127,8 @@ void visit_lqp(const std::shared_ptr<Node>& lqp, Visitor visitor) {
   }
 }
 
+enum class LQPUpwardVisitation { VisitOutputs, DoNotVisitOutputs };
```

Comment:
Sneaking this in here as it does not justify its own PR

Reply:
I am not a fan of this being part of this change, but neither have I very strong feelings about it. So, OK for me.